test/absinthe/execution/subscription_test.exs

```diff
@@ -79,6 +79,47 @@ defmodule Absinthe.Execution.SubscriptionTest do
     end
   end
 
+  defmodule PubSubWithDocsetRunner do
+    @behaviour Absinthe.Subscription.Pubsub
+
+    def start_link() do
+      Registry.start_link(keys: :duplicate, name: __MODULE__)
+    end
+
+    def node_name() do
+      node()
+    end
+
+    def subscribe(topic) do
+      Registry.register(__MODULE__, topic, [])
+      :ok
+    end
+
+    def publish_subscription(topic, data) do
+      message = %{
+        topic: topic,
+        event: "subscription:data",
+        result: data
+      }
+
+      Registry.dispatch(__MODULE__, topic, fn entries ->
+        for {pid, _} <- entries, do: send(pid, {:broadcast, message})
+      end)
+    end
+
+    def publish_mutation(_proxy_topic, _mutation_result, _subscribed_fields) do
+      # this pubsub is local and doesn't support clusters
+      :ok
+    end
+
+    def run_docset(pubsub, docs_and_topics, _mutation_result) do
+      for {topic, _key_strategy, _doc} <- docs_and_topics do
+        # publish mutation results to topic
+        pubsub.publish_subscription(topic, %{data: %{runner: "calls the custom docset runner"}})
+      end
+    end
```

Comment:
Here our pubsub behavior implements the optional run_docset callback and publishes static data to each topic.